src/components/App.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import NavBar from './navigation/NavBar';
+import Main from './editor/Main';
+import Editor from './editor/Editor';
+
+const mapStateToProps = (state) => {
+  const { pageId, sectionId } = state.location.payload;
+  return {
+    displayMain: !pageId,
+    displayPage: pageId && !sectionId,
+    displaySection: pageId && sectionId,
+    pageId,
+    sectionId,
+  };
+};
+
+const App = ({
+  displayMain = true,
+  displayPage = false,
+  displaySection = false,
+  pageId = '',
+  sectionId = '',
+}) => (
+  <div className="app">
+    <NavBar type="page" selectedPage={pageId} />
+    {!displayMain && <NavBar type="section" selectedPage={pageId} selectedSection={sectionId} />}
+    {displayMain && <Main />}
+    {displayPage && <Editor type="page" pageId={pageId} />}
```

Comment:
We could save some lines by using `displayType` ("section" or "page") and always passing in both `pageId` and `sectionId` (doesn't matter if the latter is `undefined`)